client/src/client.rs

```diff
@@ -462,6 +470,30 @@ impl Client {
         self.get_config(&GetConfiguration::Value)
             .wrap_err("Failed to get configuration value")
     }
+
+    /// Gets network status seen from the peer
+    /// # Errors
+    /// Fails if sending request or decoding fails
+    pub fn get_status(&self) -> Result<serde_json::Value> {
+        let resp = http_client::get::<_, Vec<(&str, &str)>, _, _>(
+            url(&self.status_url, uri::STATUS),
+            Bytes::new(),
+            vec![],
+            self.headers.clone(),
+        )?;
+        if resp.status() != StatusCode::OK {
+            return Err(eyre!(
+                "Failed to get status with HTTP status: {}. {}",
+                resp.status(),
+                std::str::from_utf8(resp.body()).unwrap_or(""),
+            ));
+        }
+        serde_json::from_slice(resp.body()).wrap_err("Failed to decode body")
+    }
+}
+
+fn url<L: Display, R: Display>(lhs: L, rhs: R) -> String {
+    format!("{}/{}", lhs, rhs)
 }
```

Comment:
I think it is unnecessary the `format` there is quite brief and also easy to understand